system/HTTP/Exceptions/RedirectException.php

```diff
@@ -12,17 +12,71 @@
 namespace CodeIgniter\HTTP\Exceptions;
 
 use CodeIgniter\Exceptions\HTTPExceptionInterface;
+use CodeIgniter\HTTP\ResponsableInterface;
+use CodeIgniter\HTTP\ResponseInterface;
+use Config\Services;
 use Exception;
+use InvalidArgumentException;
+use LogicException;
+use Throwable;
 
 /**
  * RedirectException
  */
-class RedirectException extends Exception implements HTTPExceptionInterface
+class RedirectException extends Exception implements ResponsableInterface, HTTPExceptionInterface
 {
     /**
      * HTTP status code for redirects
      *
      * @var int
      */
     protected $code = 302;
+
+    protected ?ResponseInterface $response = null;
+
+    /**
+     * @param ResponseInterface|string $message
```

Comment:
```suggestion
     * @param ResponseInterface|string $message a named route, relative URI, or
     *                                          a complete URL if it is a string.
     * @param int                      $code    HTTP status code for the redirect.
```

Reply:
A named route does not work from v4.0.0. See #7628

Reply:
I agree that the field description is needed, but the string can only be a relative URI.
ResponseInterface inherits from MessageInterface, so the variable name matches implicitly. )

How about this argument description?
```
$message Response object or a string containing a relative URI.
$code       HTTP status code to redirect if $message is a string.
```

Reply:
Okay, it seems good and concise! Thanks.

Reply:
> but the string can only be a relative URI. (Since the string is processed by the base_url() function, the address starting with app.baseUrl will be returned.)

Yes. I sent a PR to fix the user guide: #7633